src/Interpreters/Access/InterpreterCheckGrantQuery.cpp

```diff
@@ -0,0 +1,79 @@
+#include <Interpreters/InterpreterFactory.h>
+#include <Interpreters/Access/InterpreterCheckGrantQuery.h>
+#include <Interpreters/executeQuery.h>
+#include <Parsers/Access/ASTCheckGrantQuery.h>
+#include <Parsers/Access/ASTRolesOrUsersSet.h>
+#include <Access/AccessControl.h>
+#include <Access/ContextAccess.h>
+#include <Access/Role.h>
+#include <Access/RolesOrUsersSet.h>
+#include <Access/User.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/removeOnClusterClauseIfNeeded.h>
+#include <Interpreters/QueryLog.h>
+#include <Interpreters/executeDDLQueryOnCluster.h>
+#include <boost/range/algorithm/copy.hpp>
+#include <boost/range/algorithm/set_algorithm.hpp>
+#include <boost/range/algorithm_ext/erase.hpp>
+#include <Interpreters/DatabaseCatalog.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Processors/Sources/SourceFromSingleChunk.h>
+#include "Storages/IStorage.h"
+
+namespace DB
+{
+
+namespace
+{
+    /// Extracts access rights elements which are going to check grant
+    void collectAccessRightsElementsToGrantOrRevoke(
+        const ASTCheckGrantQuery & query,
+        AccessRightsElements & elements_to_check_grant)
+    {
+        elements_to_check_grant.clear();
+        /// GRANT
+        elements_to_check_grant = query.access_rights_elements;
+    }
+}
+
+
+BlockIO InterpreterCheckGrantQuery::execute()
+{
+    auto & query = query_ptr->as<ASTCheckGrantQuery &>();
+    query.access_rights_elements.eraseNonGrantable();
+
+    auto current_user_access = getContext()->getAccess();
+
+    /// Collect access rights elements which will be checked.
+    AccessRightsElements elements_to_check_grant;
+    collectAccessRightsElementsToGrantOrRevoke(query, elements_to_check_grant);
```

Comment:
These two lines can be replaced with just one:
```suggestion
    const AccessRightsElements & elements_to_check_grant = query.access_rights_elements;
```
and function `collectAccessRightsElementsToGrantOrRevoke` is not needed in this file.